source/daplink/interface/main.c

```diff
@@ -363,6 +366,29 @@ void main_task(void * arg)
                 reset_pressed = 0;
             }
 
+#ifdef PBON_BUTTON
+            // handle PBON pressed
```

Comment:
I've seen some comments about `PBON_BUTTON`  - why is this needed here and how this can be used generically ?

Reply:
targets with Power button (like musca) can #define PBON_BUTTON in there override file and power button handling will be compiled for those boards.

Reply:
I read the comments in previous PR -is this really needed ? If I remove this ifdef from here - what is a result? Why there is this button check, it's not common funcitonality.

Even with ifdef, this is polluting common code. I would rather have a hook - functionality there to hook into than ifdef. I merged this to progress , but still see this as unresolved